python/dgl/nn/pytorch/sparse_emb.py

```diff
@@ -376,6 +376,36 @@ def all_set_embedding(self, values):
         if th.distributed.is_initialized():
             th.distributed.barrier()
 
+    def _all_get_tensor(self, shared_name, tensor, shape):
+        """A helper function to get model-parallel tensors."""
+        # create a shared memory tensor
+        if self._rank == 0:
+            # root process creates shared memory
+            val = create_shared_mem_array(
+                shared_name,
+                shape,
+                tensor.dtype,
+            )
+            self._store.set(shared_name, shared_name)
+        else:
+            self._store.wait([shared_name])
```

Comment:
Will this blocked when the case is not multi-gpu?

Reply:
This function is abstracted from https://github.com/dmlc/dgl/blob/45153fc07833c84fd0bd8fcdeaf4bd4e796b8d7c/python/dgl/nn/pytorch/sparse_emb.py#L398-L426 so that it can be reused to gather optimizer states.

It's an internal helper function that should only be called in multi-GPU DDP training. The single-GPU and CPU cases are routed in upper level functions such as `all_get_embedding()` and `_all_get_optm_state()`. I'll update the docstring.

Reply:
Got you, it make sense if the case is just for DDP.